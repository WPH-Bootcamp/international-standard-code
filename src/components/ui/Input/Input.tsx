import React from 'react';
import type { InputProps } from './types';

const Input: React.FC<InputProps> = (props) => {
  return <input {...props} />;
};

export default Input;
