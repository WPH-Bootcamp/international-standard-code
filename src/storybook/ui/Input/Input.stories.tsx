import Input from '../../../components/ui/Input';

export default {
  title: 'Components/Input',
  component: Input,
  tags: ['autodocs'],
  argTypes: {
    type: {
      control: 'text',
      description: 'Type of input (text, password, email, etc.)',
    },
    placeholder: {
      control: 'text',
      description: 'Text shown when the input is empty',
    },
    disabled: {
      control: 'boolean',
      description: 'Disables the input field',
    },
    value: {
      control: 'text',
      description: 'Current text value inside the input',
    },
    onChange: {
      action: 'changed',
      description: 'Event fired when the input value changes',
    },
  },
};

export const Default = {
  args: {
    type: 'text',
    placeholder: 'Enter text...',
  },
};

export const Password = {
  args: {
    type: 'password',
    placeholder: 'Enter password...',
  },
};

export const Disabled = {
  args: {
    type: 'text',
    placeholder: 'Disabled input',
    disabled: true,
  },
};
