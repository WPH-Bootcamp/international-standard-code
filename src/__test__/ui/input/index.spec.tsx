import Input from '../../..//components/ui/Input';
import { render } from '@testing-library/react';

describe('input component', () => {
  it('render input component', () => {
    const { container } = render(<Input type="text" />);
    expect(container).toMatchSnapshot();
  });
});
