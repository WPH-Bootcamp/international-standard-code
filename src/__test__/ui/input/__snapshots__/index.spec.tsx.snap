// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`input component render input component 1`] = `
<div>
  <input
    type="text"
  />
</div>
`;
